### DISCUSSION TITLE
Gallery example: pynecone itself

### DISCUSSION THREAD

**Discussion by gsal** [reflex-dev/reflex#533] [Q&A] [answered]
Say, in Pynecone home page, it is clearly indicated that the entire website was written in pynecone itself...

...may I take a peak at the sources?  

I am particularly interested in seeing how the .../docs/* part is done, specifically:
- having on the left a frame that never goes away and contains some kind of table-of-contents; then, 
- on the right, contents that changes according to what it was click on the left

**Comment by Alek99** [comment, marked as answer]
https://github.com/pynecone-io/pcweb

**Reply by gsal** [reply]
Oh, wow, a lot more complicated than I expected...gonna start taking stuff out until I get to a minimal working skeleton, see if I can digest it then. 

Thanks for the link.

**Reply by Alek99** [reply]
Yeah sorry its really big has all our docs, examples etc. We do have a channel on our discord for the website feel free to join and I can help if any problems occur

**Comment by gsal** [comment]
For the record, for anyone interested...here is my experience.

First, things did not seem to work with Rocky9,Python 3.9...is this true?   I did find something I had never seen before in the sources, namely, some kind of vertical line operator. I googled it and apparently I needed Python 3.11 for that vertical line to be interpreted as some kind of "Union" ....so, installed Python 3.11

The pcweb repo requirements does not include pynecone itself; so, after installing the requirements, and then pynecone as an afterthought,  I got the following error message: 
```
Because no versions of pynecone match >0.1.16,<0.2.0
 and pynecone (0.1.16) depends on httpx (>=0.23.1,<0.24.0), pynecone (>=0.1.16,<0.2.0) requires httpx (>=0.23.1,<0.24.0).
And because googletrans (3.0.0) depends on httpx (0.13.3)
 and no versions of googletrans match >3.0.0,<4.0.0, pynecone (>=0.1.16,<0.2.0) is incompatible with googletrans (>=3.0.0,<4.0.0).
So, because pcweb depends on both googletrans (^3.0.0) and pynecone (^0.1.16), version solving failed.
```

Then, I started over, this time, installing pynecone first and THEN all the other requirements...I was not able to install googletrans: 

```
Because no versions of googletrans match >3.0.0,<4.0.0
 and googletrans (3.0.0) depends on httpx (0.13.3), googletrans (>=3.0.0,<4.0.0) requires httpx (0.13.3).
And because pynecone (0.1.16) depends on httpx (>=0.23.1,<0.24.0)
 and no versions of pynecone match >0.1.16,<0.2.0, googletrans (>=3.0.0,<4.0.0) is incompatible with pynecone (>=0.1.16,<0.2.0).
So, because pcweb depends on both pynecone (^0.1.16) and googletrans (^3.0.0), version solving failed.
```

I should indicate that, when it came to installing requirements, I did not specify any of the versions...I just let it pick whatever. 

Lastly, I removed the Translator example and, somehow, I also had to remove additional stuff related to react-confetti stuff....probably because I had also removed the entire "Advanced Guide" section because of something else, I can't recall now. 

Finally, I do get a running session :-D   Now, I just need to continue removing more and more stuff, until I have a minimal example that would allow me to better understand the mechanics and morph it to my needs.
